### PR TITLE
SSF + Payment Element reboot

### LINKED ISSUE #141: Re-create the extension with `solidus extension solidus_stripe`
- [x] Blocked by #140 
- [x] Make sure the README explains what's happening (see #135) and has links to the branch in #140
- [ ] Copy over the SSF testing strategy that was created in solidus_paypal_commerce_platform (see https://github.com/solidusio/solidus_paypal_commerce_platform/pull/166) see #146

### LINKED ISSUE #142: Integrate Payment Element
We want to start from the [Payment Element](https://stripe.com/docs/payments/payment-element) which comes with support for multiple payment methods, while still working well in the classic step-based checkout featured in SSF (vs. switching to the fully external Stripe Checkout).

We need to setup the simplest possible integration and document all the steps required for setting up the development environment and using the Stripe sandbox.

- [x] Blocked by #141 
- [ ] Spike: #147 

### Subtasks

- [x] #151
- [x] #152
- [x] #153
- [x] #154
- [x] #148
- [x] #149
- [x] #150

### After all is done

- [ ] final check on documentation 
- [ ] stripe option for solidus_installer

### PR DESCRIPTION
**Pull request by elia**
## Summary

- Fixes #141
- Sets up a baseline for further development and implementing Payment Element for SSF

## Checklist

Check out our [PR guidelines](https://github.com/solidusio/.github/blob/master/CONTRIBUTING.md#pull-request-guidelines) for more details.

The following are mandatory for all PRs:

- [ ] I have written a thorough PR description.
- [ ] I have kept my commits small and atomic.
- [ ] I have used clear, explanatory commit messages.

The following are not always needed (~cross them out~ if they are not):

- [ ] I have added automated tests to cover my changes.
- [ ] I have attached screenshots to demo visual changes.
- [ ] I have opened a PR to update the [guides](https://github.com/solidusio/edgeguides).
- [ ] I have updated the README to account for my changes.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,257 +1,30 @@
+🚧 Your looking at the source for `solidus_stripe` v5, which will only support the **starter frontend** and not the classic frontend 🚧
```
Comment (gsmendoza) [inline README.md]:
```suggestion
🚧 You're looking at the source for `solidus_stripe` v5, which will only support the **starter frontend** and not the classic frontend 🚧
```

File: app/models/solidus_stripe/gateway.rb [hunk 2]
```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+require 'stripe'
+
+# rubocop:disable Lint/UnusedMethodArgument, Style/MethodCallWithoutArgsParentheses, Lint/UselessAssignment
+module SolidusStripe
+  class Gateway
```
Comment (gsmendoza) [inline app/models/solidus_stripe/gateway.rb:7]: Regarding 189dd8746a513936412ec6181b23ddc084033dce, can you provide more details on how  Gateway, PaymentMethod, and PaymentSource were built? They seem to follow the API of SolidusPaypalCommercePlatform but their code are not exactly the same.

File: Gemfile [hunk 3]
```diff
@@ -1,38 +1,8 @@
...
 gemspec
 
-# Use a local Gemfile to include development dependencies that might not be
-# relevant for the project or for other contributors, e.g. pry-byebug.
-#
-# We use `send` instead of calling `eval_gemfile` to work around an issue with
-# how Dependabot parses projects: https://github.com/dependabot/dependabot-core/issues/1658.
-send(:eval_gemfile, 'Gemfile-local') if File.exist? 'Gemfile-local'
+gem 'stringio'
+gem 'timeout'
```
Comment (gsmendoza) [inline Gemfile:8]: Regarding 8f043dae057d21b9b32004c58d6165cf0ad85139, the commit message mentions "`libvips` was missing from the CI", but the commit doesn't seem to do anything about `libvips`. Maybe the commit message is not up-to-date?

**Review by gsmendoza — COMMENTED** [review]
@elia Looking good! I just have a few questions :)

File: app/models/solidus_stripe/gateway.rb (hunk 2 above)
Comment (elia) [reply app/models/solidus_stripe/gateway.rb:7]: It's just meant to be a placeholder for the implementation. Given the Gateway API is mandated by Solidus I felt it was ok to stub it out letting the implementation to be added in following PRs.

File: README.md [hunk 4]
```diff
@@ -1,257 +1,30 @@
+🚧 You're looking at the source for `solidus_stripe` v5, which will only support the **starter frontend** and not the classic frontend 🚧
+🚧 You can find the source for [v4 on the corresponding branch](https://github.com/solidusio/solidus_stripe/tree/v4) 🚧
```
Comment (kennyadsl) [inline README.md]: I'd also add some accurate instructions about what to do to stick to the old version, suggesting to use the Rubygems version, or if someone wants to stick to github, they will need to change the branch.

**Review by kennyadsl — APPROVED** [review]
Left a suggestions, non blocking.

File: README.md (hunk 4 above)
Comment (elia) [reply README.md]: Added and updated the layout:

image

I also added a post install message that should better cover the case of people already using it from source.

Comment (kennyadsl) [reply README.md]: Lovely, thanks!

**Pull request merged**